### PR TITLE
Print error text in <failure> tag content for more readable junit report

### LINKED ISSUE #2452: JUnit XML should contain more information in the failure text
### Your feature request related to a problem? Please describe.

[The JUnit XML Format specifies](https://www.ibm.com/docs/en/developer-for-zos/14.1.0?topic=formats-junit-xml-format) that the failure text should contain more context than just the wrong expression. golangci-lint only shows the wrong code expression with no context.

I'm using GitLab CI and the GitLab Build UI only shows the text of a failure, ignoring the "message" attribute. Running into linting errors results in a message like this:

```
Name: gosimple
Execution time: 0.00ms
System output:
	r, err := regexp.Compile("\\d+$")
```

The corresponding XML entry is:

```xml
<failure message="S1007: should use raw string (`...`) with regexp.Compile to avoid having to escape twice"><![CDATA[	r, err := regexp.Compile("\\d+$")]]></failure>
```

Without the additional information the output in the CI system is not useful on its own. As a developer I have to check the build output or run the linter locally on my machine to see the error message or suggested fix.

### Describe the solution you'd like.

I'd be happy if the JUnit XML reports could be more verbose to also include the message in the text of the `<failure>` element. As far as I can see this is also intended in the JUnit XML specification: https://www.ibm.com/docs/en/developer-for-zos/14.1.0?topic=formats-junit-xml-format (see the example at the end)

### Describe alternatives you've considered.

Scrolling through build outputs or running golangci-lint locally.

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by byrnedo**
Just adds the actual lint error text that is currently only in the `message` attribute, into the content of the `<failure>` tag.
Fixes #2452

### CODE REVIEW

File: pkg/printers/junitxml.go [hunk 1]
```diff
@@ -56,8 +56,8 @@ func (JunitXML) Print(ctx context.Context, issues []result.Issue) error {
 			Name:      i.FromLinter,
 			ClassName: i.Pos.String(),
 			Failure: failureXML{
-				Message: i.Text,
-				Content: strings.Join(i.SourceLines, "\n"),
+				Message: i.Pos.String() + ": " + i.Text,
+				Content: i.Pos.String() + ": " + i.Text + "\n" + strings.Join(i.SourceLines, "\n"),
```
Comment (ldez) [inline pkg/printers/junitxml.go:60]:
```suggestion
				Type:    i.Severity,
				Message: i.Pos.String() + ": " + i.Text,
				Content: fmt.Sprintf("%s: %s\nCategory: %s\nFile: %s\nLine: %d\nDetails: %s",
					i.Severity, i.Text, i.FromLinter, i.Pos.Filename, i.Pos.Line, strings.Join(i.SourceLines, "\n")),
```

**Review by ldez — CHANGES_REQUESTED** [review]
The page [here](https://www.ibm.com/docs/en/developer-for-zos/14.1.0?topic=formats-junit-xml-format) provides an example:
```xml
<?xml version="1.0" encoding="UTF-8" ?> 
   <testsuites id="20140612_170519" name="New_configuration (14/06/12 17:05:19)" tests="225" failures="1262" time="0.001">
      <testsuite id="codereview.cobol.analysisProvider" name="COBOL Code Review" tests="45" failures="17" time="0.001">
         <testcase id="codereview.cobol.rules.ProgramIdRule" name="Use a program name that matches the source file name" time="0.001">
            <failure message="PROGRAM.cbl:2 Use a program name that matches the source file name" type="WARNING">
WARNING: Use a program name that matches the source file name
Category: COBOL Code Review – Naming Conventions
File: /project/PROGRAM.cbl
Line: 2
      </failure>
    </testcase>
  </testsuite>
</testsuites>
```

the struct `failureXML` must be changed like that:

```go
type failureXML struct {
	Message string `xml:"message,attr"`
	Type    string `xml:"type,attr"`
	Content string `xml:",cdata"`
}
```

File: pkg/printers/junitxml.go (hunk 1 above)
Comment (byrnedo) [reply pkg/printers/junitxml.go]: @ldez That's done now, pushed

**Review by ldez — APPROVED** [review]

**Comment by byrnedo** [comment]
@ldez any idea when this will get released?

**Pull request merged**